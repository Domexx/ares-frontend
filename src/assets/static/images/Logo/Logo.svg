<svg width="197" height="73" viewBox="0 0 197 73" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect width="197" height="73" fill="url(#pattern0)"/>
<defs>
<pattern id="pattern0" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_285_662" transform="scale(0.00507614 0.0136986)"/>
</pattern>
<image id="image0_285_662" width="197" height="73" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMUAAABJCAYAAACNSuJTAAAACXBIWXMAAAsTAAALEwEAmpwYAAAGAGlUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPD94cGFja2V0IGJlZ2luPSLvu78iIGlkPSJXNU0wTXBDZWhpSHpyZVN6TlRjemtjOWQiPz4gPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iQWRvYmUgWE1QIENvcmUgNS42LWMxNDUgNzkuMTYzNDk5LCAyMDE4LzA4LzEzLTE2OjQwOjIyICAgICAgICAiPiA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPiA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtbG5zOmRjPSJodHRwOi8vcHVybC5vcmcvZGMvZWxlbWVudHMvMS4xLyIgeG1sbnM6cGhvdG9zaG9wPSJodHRwOi8vbnMuYWRvYmUuY29tL3Bob3Rvc2hvcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RFdnQ9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZUV2ZW50IyIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ0MgMjAxOSAoTWFjaW50b3NoKSIgeG1wOkNyZWF0ZURhdGU9IjIwMTYtMDktMDNUMTU6MTM6NTYrMDI6MDAiIHhtcDpNb2RpZnlEYXRlPSIyMDE5LTA0LTA0VDIzOjM4OjA3KzAyOjAwIiB4bXA6TWV0YWRhdGFEYXRlPSIyMDE5LTA0LTA0VDIzOjM4OjA3KzAyOjAwIiBkYzpmb3JtYXQ9ImltYWdlL3BuZyIgcGhvdG9zaG9wOkNvbG9yTW9kZT0iMyIgcGhvdG9zaG9wOklDQ1Byb2ZpbGU9InNSR0IgSUVDNjE5NjYtMi4xIiB4bXBNTTpJbnN0YW5jZUlEPSJ4bXAuaWlkOjE0MjdjMTUxLTg0MDUtNGQyMy04MTNiLTFjOTcyMjgzMjkzMCIgeG1wTU06RG9jdW1lbnRJRD0iYWRvYmU6ZG9jaWQ6cGhvdG9zaG9wOmQ5MTI3YzZjLTUzMDEtNmQ0Ny1iYzJjLTQ3OWFhY2IxOGFlMyIgeG1wTU06T3JpZ2luYWxEb2N1bWVudElEPSJ4bXAuZGlkOjkwODE1OWMzLTAzMDEtNDkwNC04ZDIxLTYzYjUyN2NmYWNmNiI+IDx4bXBNTTpIaXN0b3J5PiA8cmRmOlNlcT4gPHJkZjpsaSBzdEV2dDphY3Rpb249ImNyZWF0ZWQiIHN0RXZ0Omluc3RhbmNlSUQ9InhtcC5paWQ6OTA4MTU5YzMtMDMwMS00OTA0LThkMjEtNjNiNTI3Y2ZhY2Y2IiBzdEV2dDp3aGVuPSIyMDE2LTA5LTAzVDE1OjEzOjU2KzAyOjAwIiBzdEV2dDpzb2Z0d2FyZUFnZW50PSJBZG9iZSBQaG90b3Nob3AgQ0MgMjAxOSAoTWFjaW50b3NoKSIvPiA8cmRmOmxpIHN0RXZ0OmFjdGlvbj0ic2F2ZWQiIHN0RXZ0Omluc3RhbmNlSUQ9InhtcC5paWQ6MTQyN2MxNTEtODQwNS00ZDIzLTgxM2ItMWM5NzIyODMyOTMwIiBzdEV2dDp3aGVuPSIyMDE5LTA0LTA0VDIzOjM4OjA3KzAyOjAwIiBzdEV2dDpzb2Z0d2FyZUFnZW50PSJBZG9iZSBQaG90b3Nob3AgQ0MgMjAxOSAoTWFjaW50b3NoKSIgc3RFdnQ6Y2hhbmdlZD0iLyIvPiA8L3JkZjpTZXE+IDwveG1wTU06SGlzdG9yeT4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz4HG/yNAAAEpUlEQVR4nO2d3XHiMBSFL5kUkifTwLbBqzM0kAK8dSwFbAF4llfaSAPLUwrYHrwviIiLZBuhnyNzvhlPwLGtD0vHkmVmWMk3G8HlKOX8SpatQXPRlKyjubgcr/Zf2Rs2bbcNlErG6bDrzevcfiXLrsVFVFBL1pFMh+OmnVv7X/Z9tTc8HXZwoWja7vI6t1/JsjWILucGddXQStWR5eILxsXz759fF8e1Z9+NiGyattuLyIC2IHghOAC7QLQd4yLu4dGljQ/DMDRttzeLea/3hQ4FF+wFqd14gnETiMHCFYxX8TB8+f4zzert8ePYx9A84oZetgbdxTdcKlFHp8Nu27TdnKGUiIis33/2rvUvEfyyMXzlaQi5yplDbS6l6+gcDBNU72zY+v1nb99b2FQTihINA6UxitThglJHKhhmXW+CYHoI+709i+UdPhFSM2ooZdb1axG5mn1SgRCpJBQlr5LD1/i4PifILoh1NBYM+73a7VhFKAgJxRcMz+ZHkYruKQgJxXWP4eAyU8VQkMUxfNyumwjG1dQtQ0EWycxgHMXxLIOhIItlRjCczzGeIhQoMzarN7r4SOUSEozFh8Kc7GdoACEguKSuo3uDsfhQECLiDoYPhoI8DToYvt6CoSBPxZweg6EgRMFQEKJgKAhRJPlCINJ3/wm5F/YUBUC6aCC5oMCvjmcEqQEiuaDBnqIiEJ4+G5BcYsNQEKLg8KkCkK7KSC6p8IfiX0YL4gSpASK5pKaOnuKBgA6fcY4T9RixWKILwGfiPQUww6cKdUGQXFLDUFQAUoNEckkFQ0GIgqGoCKQrNJJLbBgKQhRJZp9WP75fL/mKci/mvCCcEyQXNNhTFMC+aJQGyQUFhoIQBUNBiKKOJ9oPwPsbci/sKQhRMBSEKBgKQhQMBSEKhqIASDf8CC7GAcFF5Almn5BAqXQRLBcRLB/2FIQoGApCFAwFIQqGoiKQvrynXUq6xS6boSBEwVBUAmovYf8ISglHXeY9P+Plg1Oy4KCGYWqb1FOsKc+LNxSxCo1xnCW6hDJ8iKx+47iYv8bJUMItRi8hwuFTVcSq9Bhol9JuMcvn8KkCSjc4mzEXV4+RmhTn5hKK02HXN20np8Nua28QWqg+OSHHiXEMNJcYuBofSnCQ3eZyNXw6B6O314UmX5+IkOPEOAaaSyzQfGyQ3ebwIiJHewWDkd4lFmg+NshuU5iegsHI7BILNB8bZLcx7OETg5HZJRZoPmMguxn0lCyDkdklFmg+Yw4l3eaU7XpOwWBkdokFms8YCG5N2/Wnw67X630P7xiMzC4xcJVdykeX27RdtDYUytzyVhP/39hvmrbb8jlGWpcQ5lR2Lh+Xi31FdrUhkbR+U05njrdb+dnYS9N2exEZuOAuTdvtUerp7AHVhlxOEgDUh+Iyr8JL15On8RV1mxOIqeGTzeRQipTDjNf1jWOJevK5OLbbnrdL7jfidDNsuicUIo5g3Lk/ScRYA8xdTx4X0/hursw5/Cacrrg3FCKB4y/y1OjGh9CGvDfWIaEQwfhQxI/3qlwAX+Mr6TY60xQaChGME06ucVV2qXqaO8WZ02+W0389pyV1yiU5YQAAAABJRU5ErkJggg=="/>
</defs>
</svg>
