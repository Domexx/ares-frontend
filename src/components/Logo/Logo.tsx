import Image from 'next/image';
import React from 'react';
import s from './Logo.module.css';

export type Props = {
  url?: string;
  layout?: 'fixed' | 'fill' | 'intrinsic' | 'responsive';
  width?: number;
  height?: number;
};

const Logo: React.FC<Props> = ({ url = '#', layout = 'fixed', width = 122, height = 53 }) => {
  return (
    <div className={s['logo']}>
      <a href={url}>
        <Image src="/static/images/Logo/logo.gif" width={width} height={height} layout={layout} alt="Cosmic_Logo" />
      </a>
    </div>
  );
};

export default Logo;
