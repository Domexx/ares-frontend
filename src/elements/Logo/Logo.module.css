/* === === === === Logo === === === === */

.Logo {
  margin-top: 0;
}
