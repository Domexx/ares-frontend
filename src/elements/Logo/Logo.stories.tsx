import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import LogoComponent, { Props } from './Logo';

const Component = {
  title: 'Elements/Logo',
  component: LogoComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof LogoComponent>> = (args) => {
  return <LogoComponent {...args} />;
};

export const Logo = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
Logo.args = {} as Props;
