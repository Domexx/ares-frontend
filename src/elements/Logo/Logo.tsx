import React from 'react';
import s from './Logo.module.css';

import { ReactComponent as HabboLogo } from '../../assets/static/images/Logo/Logo.svg';

export type Props = {
  url?: string;
};

const Logo: React.FC<Props> = ({ url = '#' }) => {
  return (
    <div className={s['Logo']}>
      <a href={url}>
        <HabboLogo />
      </a>
    </div>
  );
};

export default Logo;
