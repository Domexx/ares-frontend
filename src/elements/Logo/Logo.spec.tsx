import React from 'react';

import { render } from '@testing-library/react';

import Logo from './Logo';

describe('Logo Component', () => {
  it('should render properly', () => {
    const { container } = render(<Logo />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
