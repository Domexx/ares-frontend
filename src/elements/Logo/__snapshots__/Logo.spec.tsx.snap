// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component should render properly 1`] = `
<div
  class="logo"
>
  <a
    href="#"
  >
    <svg />
  </a>
</div>
`;
